### PR TITLE
Allow Arc to deal with final methods of beans that need to be proxied

### LINKED ISSUE #10290: RequestScoped beans keeping property values from previous requests when using kotlin.
**Describe the bug**
When using kotlin and injecting a RequestScoped bean, on each new request a new bean instance is initialized, but for some reason, the property values from the previous request bean are kept, this seems to be an issue only with the kotlin extension, because I couldn't reproduce the issue using Java.

**Expected behavior**
The new request bean should not contain previous property values

**Actual behavior**
A new bean instance is created on each request, but the same property values from previous requests are propagated.

**To Reproduce**
I've created a project to reproduce the issue here: [https://github.com/leandrobortoli/quarkus-issue-request-scoped](https://github.com/leandrobortoli/quarkus-issue-request-scoped)

On each request, it checks if the bean property has been initialized and if not initializes it with a random value, as a result, we can see that only the first request initializes the properties and the values are propagated in the subsequent requests.

```
@Path("/ping")
class PingController {

    @Inject
    lateinit var requestScopedBean: RequestScopedBean

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun ping(): String {
        if(requestScopedBean.myProperty == null) {
            println("Setting requestScopedBean.myProperty")
            requestScopedBean.myProperty = java.util.Random().nextInt()
        }
        if(requestScopedBean.myStringProperty == null) {
            println("Setting requestScopedBean.myStringProperty")
            requestScopedBean.myStringProperty = StringUtils.randomString()
        }
        println("requestScopedBean ref: $requestScopedBean")
        println("requestScopedBean.myProperty: ${requestScopedBean.myProperty}")
        println("requestScopedBean.myStringProperty: ${requestScopedBean.myStringProperty}")
        return "pong"
    }
}
```

```
@RequestScoped
open class RequestScopedBean {

    var myProperty: Int? = null
    var myStringProperty: String? = null

    init {
        println("Initializing RequestScopedBean")
    }
}
```

**Screenshots**
OUTPUT WITH 3 CONSECUTIVE REQUESTS:
image

**Environment (please complete the following information):**
 - Microsoft Windows [Version 10.0.18362.900]
 -Java version "1.8.0_162"
Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)
 - Quarkus version 1.5.2.Final
 - Gradle 6.4.1

### PR DESCRIPTION
**Pull request by geoand**
This is done in the same manner that is already present for
interceptors and is very useful for Kotlin code where methods
are final by default

Fixes: #10290

### CODE REVIEW

**Comment by geoand** [comment]
@mkouba this probably could be a little cleaner, but I hope it's good enough for now :)

**Review by gastaldi — APPROVED** [review]
LGTM, but let's wait for @mkouba to review :)

**Review by mkouba — COMMENTED** [review]
Looks good. 

I wonder if we should mention this in the guide (under the "Non-standard Features" section) and possibly also mention the fact that we do this for beans classes as well. Maybe even enhance the javadoc of `ArcConfig.transformUnproxyableClasses`?

**Comment by geoand** [comment]
> Looks good.
> 
> I wonder if we should mention this in the guide (under the "Non-standard Features" section) and possibly also mention the fact that we do this for beans classes as well. Maybe even enhance the javadoc of `ArcConfig.transformUnproxyableClasses`?

Sure, I'll do that later on

**Comment by geoand** [comment]
@mkouba PR updated with the doc suggestions you mentioned

**Comment by geoand** [comment]
@mkouba OK to merge?

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java [hunk 1]
```diff
@@ -59,6 +59,9 @@
      * If set to true, the bytecode of unproxyable beans will be transformed. This ensures that a proxy/subclass
      * can be created properly. If the value is set to false, then an exception is thrown at build time indicating that a
      * subclass/proxy could not be created.
+     *
+     * Furthermore when enabled, this setting allows Quarkus to properly create regular scoped class beans whose methods
```
Comment (mkouba) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java:63]: I was more thinking about a list of all transformations we do... final class, no-args constructors, final methods of bean classes (for proxies/subclasses). This sentence is more or less redundant.

Comment (geoand) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java]: OK, fixed

**Review by mkouba — APPROVED** [review]
LGTM.

**Pull request merged**